### PR TITLE
Merge in examples/kdbset.c

### PR DESCRIPTION
**Pull request by Chemin1**
Depends on #3234
Helps with #3131

The call to elektraMerge should be ok. I am still a bit confused by this example.
@markus2330 Is this what you imagined [here](https://github.com/ElektraInitiative/libelektra/issues/3131#issuecomment-554933467)?

### CODE REVIEW

File: examples/kdbset.c [hunk 1]
```diff
@@ -51,42 +52,43 @@ kdbGet (handle, myConfig, parentKey); // kdbGet needs to be called first!
...
-		// should always work, we just write what we got
-		// but to be sure always give the user another way
-		// to exit the loop
-		kdbGet (handle, myConfig, parentKey); // refresh key database
-		break;
-		// other cases ...
+	int strategy = showElektraErrorDialog (parentKey);
+	theirs = ksDup (ours);
+	kdbGet (handle, theirs, parentKey); // refresh key database
+	Key * informationKey = keyNew(0, KEY_END);
```
Comment (markus2330) [inline examples/kdbset.c]: Is this needed? Or can we also pass the parentKey there?

File: examples/kdbset.c [hunk 2]
```diff
@@ -51,42 +52,43 @@ kdbGet (handle, myConfig, parentKey); // kdbGet needs to be called first!
...
+	KeySet * res = elektraMerge(
```
Comment (markus2330) [inline examples/kdbset.c]:
```suggestion
	KeySet * result = elektraMerge(
```

I think every developer has these few bytes available on the hard disc.

File: examples/kdbset.c [hunk 3]
```diff
@@ -51,42 +52,43 @@ kdbGet (handle, myConfig, parentKey); // kdbGet needs to be called first!
...
+	} else {
+		// an error happened while merging
+		if (numberOfConflicts > 0 && strategy == MERGE_STRATEGY_ABORT)
+		{
+			// Error due to merge conflicts
+			ret = -1;
+		}
+		else
+		{
+			// Other error
```
Comment (markus2330) [inline examples/kdbset.c]: What could this be?

File: examples/kdbset.c (hunk 1 above)
Comment (Chemin1) [reply examples/kdbset.c]: The key will only get some metadata. I changed it to parentKey.

**Comment by Chemin1** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Please rebase.

**Comment by Chemin1** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Thank you, so nice to have these neat example.

**Pull request merged**